Source/Dafny/DafnyOptions.cs

```diff
@@ -827,7 +827,25 @@ public enum IncludesModes {
       TODO
 
     {:no_inline}
-      TODO
+      When functions such as `function f(x: int) { x % 2 == 0 }`
+      are used in assertions like `assert f(6);`, Dafny
+      will by default try to figure out if it can split the call
+      into multiple assertions that are easier for the verifier.
+      Hence, sometimes, if sound to do so (e.g. no `{:opaque}`),
+      Dafny will inline the function, resulting in, for example,
+      `assert 6 % 2 == 0`.
+      
+      Adding the attribute `{:no_inline}` to a function will result
+      in the Dafny verifier to never inline the function in such
+      a function call, but unless the function is `{:opaque}`, its
+      definition will still be available.
+      
+      This trick can be helpful if for a huge conjunct predicate `P`,
```

Comment:
I don't understand this example. How do P, P(x) and f(x) relate?

Reply:
> I don't understand this example. How do P, P(x) and f(x) relate?

Fixed. It should have been all P